src/Maybe.js

```diff
@@ -29,11 +35,19 @@ Maybe.of = Maybe.Just;
 Maybe.prototype.of = Maybe.Just;
 
 Maybe.isJust = function(x) {
-  return x instanceof _Just;
+  if (typeof x.isJust !== 'boolean') {
```

Comment:
Would you mind doing the same in the `Either` PR #86? Then I think we're in a good position for both of them. :)